libs/sql-parser/src/main/java/io/crate/sql/tree/BitString.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.sql.tree;
+
+import java.util.BitSet;
+
+public class BitString extends Literal implements Comparable<BitString> {
+
+    public static BitString ofBitString(String bitString) {
+        assert bitString.startsWith("B'") : "Bitstring must start with B'";
+        return ofRawBits(bitString.substring(2, bitString.length() - 1));
```

Comment:
Don't we need to validate that it ends with `'` as well? I saw additional check inside `toBitSet`  but since we are focusing on enclosing symbols separately it's probably worth to cover ending as well.